test/mainnet-fork/ForkTestBase.t.sol

```diff
@@ -192,88 +176,107 @@ contract ForkTestBase is DssTest {
         AllocatorInit.initShared(dss, sharedInst);
         AllocatorInit.initIlk(dss, sharedInst, ilkInst, ilkConfig);
-
         vm.stopPrank();
 
-        /*** Step 3: Deploy ALM system ***/
+        buffer = ilkInst.buffer;
+        vault  = ilkInst.vault;
 
-        almProxy = new ALMProxy(SPARK_PROXY);
+        /*** Step 3: Deploy and configure ALM system ***/
 
-        rateLimits = new RateLimits(SPARK_PROXY);
-
-        mainnetController = new MainnetController({
-            admin_      : SPARK_PROXY,
-            proxy_      : address(almProxy),
-            rateLimits_ : address(rateLimits),
-            vault_      : ilkInst.vault,
-            psm_        : PSM,
-            daiUsds_    : usdsInst.daiUsds,
-            cctp_       : CCTP_MESSENGER,
-            susds_      : susdsInst.sUsds
+        ControllerInstance memory controllerInst = MainnetControllerDeploy.deployFull({
+            admin  : Ethereum.SPARK_PROXY,
+            vault  : ilkInst.vault,
+            psm    : Ethereum.PSM,
+            daiUsds: Ethereum.DAI_USDS,
+            cctp   : Ethereum.CCTP_TOKEN_MESSENGER,
+            susds  : Ethereum.SUSDS
         });
 
+        almProxy          = ALMProxy(payable(controllerInst.almProxy));
+        rateLimits        = RateLimits(controllerInst.rateLimits);
+        mainnetController = MainnetController(controllerInst.controller);
+
         CONTROLLER = almProxy.CONTROLLER();
         FREEZER    = mainnetController.FREEZER();
         RELAYER    = mainnetController.RELAYER();
 
-        /*** Step 4: Configure ALM system in allocation system ***/
-
-        vm.startPrank(SPARK_PROXY);
+        MainnetControllerInit.AddressParams memory addresses = MainnetControllerInit.AddressParams({
+            admin         : Ethereum.SPARK_PROXY,
+            freezer       : freezer,
+            relayer       : relayer,
+            oldController : address(0),
+            psm           : Ethereum.PSM,
+            vault         : vault,
+            buffer        : buffer,
+            cctpMessenger : Ethereum.CCTP_TOKEN_MESSENGER,
+            dai           : Ethereum.DAI,
+            daiUsds       : Ethereum.DAI_USDS,
+            usdc          : Ethereum.USDC,
+            usds          : Ethereum.USDS,
+            susds         : Ethereum.SUSDS
+        });
 
-        IVaultLike(ilkInst.vault).rely(address(almProxy));
+        RateLimitData memory usdsMintData = RateLimitData({
+            maxAmount : 5_000_000e18,
+            slope     : uint256(1_000_000e18) / 4 hours
+        });
 
-        mainnetController.grantRole(FREEZER, freezer);
-        mainnetController.grantRole(RELAYER, relayer);
+        RateLimitData memory usdcToUsdsData = RateLimitData({
+            maxAmount : 5_000_000e6,
+            slope     : uint256(1_000_000e6) / 4 hours
+        });
 
-        almProxy.grantRole(CONTROLLER, address(mainnetController));
+        RateLimitData memory usdcToCctpData = RateLimitData({
```

Comment:
This one can be unlimited. I think that is more likely in practise.

Reply:
Updated

Reply:
It's still limited.